### PR TITLE
Initial redux push

### PR DESCRIPTION
**Pull request by penguingovernor**
This should help out with #9. 

It currently just adds the editor's contents to the central data store on their creation. It doesn't update the data store when the editors are edited (try saying that five times fast 😜).  That'll be another PR.

### CODE REVIEW

File: src/reducer.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    case EditorChanged:
+      break;
+    case EditorAdded:
+      return produce(store, (storeDraft) => {
+        storeDraft.editors.push({
+          type: action.payload.type,
+          body: action.payload.body,
+        });
+      });
+    default:
```
Comment (wlawt) [inline src/reducer.js:15]: Does this PR also add switch cases for `ConfigType` `PluginType` `SourceType` ?

Comment (penguingovernor) [reply src/reducer.js:15]: Those are for the actual data store, do you think they could use a better name? 

Relevant: https://github.com/MLH-Fellowship/babel-sandbox/pull/34/files#diff-57aa11691094c3295e1f9dcbe8432ba2R131

Comment (wlawt) [reply src/reducer.js:15]: right, iirc having a type in a dispatch usually calls on the reducer. so in the actual data store that might just appending all of it to an object to save the state, if I'm interpreting this correctly. 

but we could also save that for a later PR, thoughts @ianjennings?

Comment (ianjennings) [reply src/reducer.js:15]: From what I understand here, this pull is only dealing with rendered "editors" and does not modify the plugin, config, source, object. It's merely for knowing what kind of `editor` we're rendering.

I think @wlawt is asking about how we'll get the data out of the editors when we need to send them to the backend.

We talked in the standup today about sending the redux state to the backend, so it sounds like @penguingovernor has a plan there.

Comment (ianjennings) [reply src/reducer.js:15]: I vote to save it for another PR

**Review by ianjennings — APPROVED** [review]

**Pull request merged**